src/main.py

```diff
@@ -0,0 +1,79 @@
+import argparse
+
+import config
+import rds
+import tf
+import utils
+
+LOGGER = utils.setup_custom_logger("root")
+
+
+def main(command_line=None):
+    parser = argparse.ArgumentParser(description="Restore RDS snapshot")
+    parser.add_argument("-c", "--config", required=True)
+    parser.add_argument("-t", "--tfstate")
+    args = parser.parse_args(command_line)
+
+    data = config.load_config(args.config)
+    config.validate(data)
+
+    if data["Target"].get("AssumeRole") is not None:
+        LOGGER.info("Assuming target AWS role")
+        data["Target"]["AssumeRole"] = utils.assume_aws_role(
+            data["Target"]["AssumeRole"]
+        )
+    else:
+        data["Target"]["AssumeRole"] = None
+
+    if data["Source"]["Share"].get("AssumeRole") is not None:
+        LOGGER.info("Assuming source AWS role")
+        data["Source"]["AssumeRole"] = utils.assume_aws_role(
+            data["Source"]["Share"]["AssumeRole"]
+        )
+    else:
+        data["Source"]["AssumeRole"] = None
```

Comment:
Same as previous, just call the variable `share_credentials` and get value from `utils.get_aws_credentials`